client/src/index.ts

```diff
@@ -346,42 +437,91 @@ export class RestRepository<T> extends Repository<T> {
         body[key] = (value as XataRecord).id;
       }
     }
-    const obj = await this.request('POST', `/tables/${this.table}/data`, body);
-    return this.client.initObject(this.table, obj);
+
+    const response = await this.request<{
+      id: string;
+      xata: { version: number };
+    }>('POST', `/tables/${this.table}/data`, body);
+    if (!response) {
+      throw new Error("The server didn't return any data for the query");
+    }
+
+    // TODO: Review this, not sure we are properly initializing the object
+    return this.client.initObject(this.table, response);
+  }
+
+  async createMany(records: T[]): Promise<T[]> {
+    // TODO: Review the id of the records
+
+    const response = await this.request<{
+      recordIDs: string[];
+    }>('POST', `/tables/${this.table}/bulk`, { records });
+    if (!response) {
+      throw new Error("The server didn't return any data for the query");
+    }
+
+    // TODO: Review this, not sure we are properly initializing the object
+    return response.recordIDs.map((record) => this.client.initObject(this.table, { id: record }));
   }
 
   async read(id: string): Promise<T | null> {
     try {
-      const obj = await this.request('GET', `/tables/${this.table}/data/${id}`);
-      return this.client.initObject(this.table, obj);
+      const response = await this.request<
+        T & { id: string; xata: { version: number; table?: string; warnings?: string[] } }
+      >('GET', `/tables/${this.table}/data/${id}`);
+      if (!response) return null;
+
+      return this.client.initObject(this.table, response);
     } catch (err) {
       if ((err as XataError).status === 404) return null;
       throw err;
     }
   }
 
   async update(id: string, object: Partial<T>): Promise<T> {
-    const obj = await this.request('PUT', `/tables/${this.table}/data/${id}`, object);
-    return this.client.initObject(this.table, obj);
+    const response = await this.request<{
+      id: string;
+      xata: { version: number };
+    }>('PUT', `/tables/${this.table}/data/${id}`, object);
+    if (!response) {
+      throw new Error("The server didn't return any data for the query");
+    }
+
+    // TODO: Review this, not sure we are properly initializing the object
+    return this.client.initObject(this.table, response);
   }
 
   async delete(id: string) {
+    // TODO: Return boolean?
     await this.request('DELETE', `/tables/${this.table}/data/${id}`);
```

Comment:
Raising this as a question of the SDK API, can this fail? Should we handle failures as exceptions or booleans? (no strong opinion)

Reply:
That's a great question. I thought about a couple of months ago and my conclusion is that we may provide two types of methods:

- `read()`/`delete()` don't throw but return null/false if something fails.
- `mustRead()`/`mustDelete()` throw if something fails. So the first one never returns null and the second one does not return a boolean.

I thought about other alternatives to the `must` prefix, but couldn't find anything better. Rails has this `!` suffix such as in `read!`, but we cannot do that.

An option that I discard is defining the behavior as a setting when creating the client. This would be error-prone and makes the typing less useful.

Reply:
In JS/TS world I hate exceptions because it's hard to let the library user they need to do a try/catch (opposite to java or other type safe languages where you declare a method can throw something).

Don't know, maybe we can leave safe methods for now and think of a naming convention for throwable methods. I do agree ``must`` seems to hard for a prefix.